### PR TITLE
API: Collaborators Page

### LINKED ISSUE #8: API: Create Collaborators contract
image
image

### PR DESCRIPTION
**Pull request by dricazenck**
Closes #8 

image
image

### CODE REVIEW

**Review by kikilondon — APPROVED** [review]

File: api_about.yml [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      parameters:
+        - in: query
+          name: memberType
+          schema:
+            type: string
+            maxLength: 30
+            pattern: '^[A-Z][a-z ]{0,30}$'
+            example: Lead
+          required: false
+          description: Filter team members by their MemberType, in case it is empty return all types
```
Comment (goelsonali) [inline api_about.yml:25]: Description can be more descriptive - Eg - Filter team members by type - director,lead, evangalist ( in order for someone to test the API with some query parameters)

File: api_about.yml [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+            leads:
+              type: array
+              maxItems: 30
+              items:
+                $ref: '#/components/schemas/Member'
+            evangelists:
+              maxItems: 6
+              type: array
+              items:
+                $ref: '#/components/schemas/Member'
```
Comment (goelsonali) [inline api_about.yml:156]: Can collaborators be another type rather than another endpoint ?

File: api_about.yml [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+          maxItems: 30
+          items:
+            $ref: '#/components/schemas/Collaborator'
+
+    Collaborator:
+      type: object
+      required:
+        - fullName
+        - position
+        - memberType
```
Comment (goelsonali) [inline api_about.yml:201]: For the collaborator do we need the "memberType" ?? if required then it is duplicate of teamPage

File: api_about.yml [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+servers:
+  - url: http://localhost:8080/api/cms/v1
+  - url: https://wcc-cms.fly.dev/api/cms/v1
+
+paths:
+  /team:
+    get:
+      summary: Get team members grouped by MemberType
+      parameters:
```
Comment (goelsonali) [inline api_about.yml:16]: FYI , I am planning to move these parameters to a common component as well - as part of Footer API ( so can then update it here as well)

File: api_home.yml [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+          type: string
+          example: Opportunities and Programmes
+        description:
+          type: string
+          example: >
+            Join our community and unlock endless opportunities. Network, find mentors, and access leadership programs. Whether you're aiming to enhance your skills, grow your professional network, or advance your career, we have what you need. We offer a wide range of opportunities to help you achieve your goals.
+        programmes:
+          type: array
+          items:
+            $ref: '#/components/schemas/Programme'
```
Comment (goelsonali) [inline api_home.yml:89]: Should be a common schema - if needs to be reused on other pages.

File: resources/home.json [hunk 6]
```diff
@@ -0,0 +1,78 @@
+{
+  "heroSection": {
```
Comment (goelsonali) [inline resources/home.json:2]: What's the pupose of these .json resources ? in openAPI service ?

File: README.md [hunk 7]
```diff
@@ -19,44 +21,45 @@ Use your editor and install the relevant plugging/extension to edit and to see t
...
+    8. The Audit Problem view page will show all the problem exist or no error will show in case of everything correct.
 
 ## How to use
 
 ### View/Edit
 
-1. Open one of the API files you want to see, for example: [api_module_about.yml](api_module_about.yml)
+1. Open one of the API files you want to see, for example: [api_module_about.yml](api_about.yml)
 2. Open the OpenAPI View from your IDE
 3. You will be able to see all paths and can start to edit
```
Comment (goelsonali) [inline README.md:64]: There are 2 servers on the swagger UI - default is set to localhost, another step should be specified here to choose the correct server.

**Review by goelsonali — COMMENTED** [review]
Added couple of comments

File: api_about.yml [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+
+    CollaboratorPage:
+      type: object
+      additionalProperties: false
+      properties:
+        page:
+          $ref: 'schemas/attributes.yml#/components/schemas/attributes/SimplePage'
+        contact:
+          $ref: 'schemas/attributes.yml#/components/schemas/attributes/Contact'
+        collaborators:
```
Comment (dricazenck) [inline api_about.yml:190]:
```suggestion
        items:
```

File: resources/home.json (hunk 6 above)
Comment (dricazenck) [reply resources/home.json:2]: It was an easy way to review, but once it is approved I can remove to avoid confusion.

File: api_home.yml (hunk 5 above)
Comment (dricazenck) [reply api_home.yml:89]: This is only for home page, the program  page will be very different from this one. So it won't be reusable.

File: api_about.yml (hunk 4 above)
Comment (dricazenck) [reply api_about.yml:16]: That is a good idea!

File: api_about.yml (hunk 3 above)
Comment (dricazenck) [reply api_about.yml:201]: I was thinking a way to one schema to have member type and the other one no. But if you know a way to hide one attribute from the schema, that would be interesting for this case.

Comment (dricazenck) [reply api_about.yml:176]: I follow your suggestions and re-used member

**Pull request merged**